docs/design/configs.md

```diff
@@ -0,0 +1,231 @@
+# Generic Configuration Object
+
+## Abstract
+
+This proposal proposes a generic configuration object named ``ConfigData`` stored inside ``etcd`` that
+stores command-line flags, environment variables, and configuration files used 
```

Comment:
@smarterclayton @pmorie @ncdc you can find how command-line flag translation will be handled in the new diff